tendermint/src/lite/error.rs

```diff
@@ -78,6 +78,10 @@ pub enum Kind {
     /// the implementation, if any.
     #[error("Implementation specific error")]
     ImplementationSpecific,
+
+    /// This is returned when a faulty i.e misbehaving full node is found
+    #[error("Faulty full node: {reason}")]
+    FaultyFullNode { reason: String },
```

Comment:
I don't think we need the `String` reason here, as anomaly lets you annotate / provide (string) context for errors.